src/filter/json_encode_filter.rs

```diff
@@ -0,0 +1,139 @@
+    /// The forwards along which the filter will emit its `metric::Event`s.
+    pub forwards: Vec<String>,
+    /// Whether the filter should attempt to parse LogLine values that are valid JSON objects.
+    pub parse_line: bool,
+}
+
+impl JSONEncodeFilter {
+    /// Create a new JSONEncodeFilter
+    pub fn new(config: &JSONEncodeFilterConfig) -> JSONEncodeFilter {
+        JSONEncodeFilter {
+            parse_line: config.parse_line,
+        }
+    }
+}
+
+impl filter::Filter for JSONEncodeFilter {
+    fn process(
+        &mut self,
+        event: metric::Event,
+        res: &mut Vec<metric::Event>,
+    ) -> Result<(), filter::FilterError> {
+        match event {
+            metric::Event::Telemetry(_) => {
+                // Drop telemetry
+            },
+            metric::Event::Log(l) => if let Some(ref log) = *l {
+                let naive_time = NaiveDateTime::from_timestamp(log.time, 0);
+                let utc_time: DateTime<Utc> = DateTime::from_utc(naive_time, Utc);
+                let metadata = json!({
+                    "time": utc_time.to_rfc3339(),
+                    "path": log.path.clone(),
+                    "tags": log.tags,
+                });
+                let metadata = metadata.as_object().unwrap();
+                // If parse_line is true, and line is parsable as a JSON object, parse it.
+                // Otherwise get an object containing the original line.
+                let value = (if self.parse_line { Some(()) } else { None })
+                    .and_then(|_| serde_json::from_str::<Value>(&log.value).ok())
+                    .and_then(|v| if let Value::Object(obj) = v { Some(obj) } else { None })
+                    .map(|v| { JSON_ENCODE_LOG_PARSED.fetch_add(1, Ordering::Relaxed); v })
+                    .unwrap_or_else(||
+                        if let Value::Object(obj) = json!({"message": log.value.clone()}) {
+                            obj
+                        } else {
+                            panic!()
+                        }
+                    );
+                // Combine our various sources of data. Metadata overrides parsed values overrides user-specified.
+                let value = merge_objects(&[&metadata, &log.fields.clone().into(), &value]);
+                res.push(metric::Event::Raw {
+                    order_by: random(),
+                    encoding: metric::Encoding::JSON,
+                    bytes: serde_json::to_string(&value).unwrap().into(), // serde_json::Value will never fail to encode
+                });
+                JSON_ENCODE_LOG_PROCESSED.fetch_add(1, Ordering::Relaxed);
+            },
+            metric::Event::TimerFlush(f) => {
+                res.push(metric::Event::TimerFlush(f));
+            }
+            raw @ metric::Event::Raw { .. } => {
+                res.push(raw);
+            }
+            metric::Event::Shutdown => {
+                res.push(metric::Event::Shutdown);
+            }
```

Comment:
It looks to me like these last 3 match arms are all doing the same thing, so this could be simplified to:
```
event => { 
    res.push(event); 
}
```

Reply:
ah, right. I just copied this part from DelayFilter.